TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/JsonEmitter.java

```diff
@@ -114,8 +111,8 @@ public ByteBuf recycleByteBufRetained() {
     private Stack<LevelContext> cursorStack;
 
     @SneakyThrows
-    public JsonEmitter() {
-        outputStream = new ImmediateByteBufOutputStream();
+    public JsonEmitter(ByteBufAllocator byteBufAllocator) {
+        outputStream = new ImmediateByteBufOutputStream(byteBufAllocator);
```

Comment:
For my own learning, can you explain why `JsonEmitter` needs a `byteBufAllocator` as a parameter?

Reply:
Before, the JsonEmitter (by way of its inner class) just used ByteBufAllocator.DEFAULT as its ByteBufAllocator.  That may or may not be the best allocator to use for this channel, context, thread, etc.  I now have the caller pass in the ChannelHandlerContext's ByteBufAllocator.

My assumption is that the allocators are probably tied to a thread (or at least they should be) & using a thread-specific one will be much faster than one shared across many threads.